starter-core/src/test/groovy/io/micronaut/starter/springboot/SpringBootStarterSpec.groovy

```diff
@@ -0,0 +1,107 @@
+package io.micronaut.starter.springboot
+
+import io.micronaut.starter.ApplicationContextSpec
+import io.micronaut.starter.BuildBuilder
+import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.build.BuildTestUtil
+import io.micronaut.starter.build.BuildTestVerifier
+import io.micronaut.starter.build.dependencies.Scope
+import io.micronaut.starter.fixture.CommandOutputFixture
+import io.micronaut.starter.options.BuildTool
+import io.micronaut.starter.options.Language
+import io.micronaut.starter.options.Options
+import io.micronaut.starter.options.TestFramework
+
+class SpringBootStarterSpec extends ApplicationContextSpec implements CommandOutputFixture {
+
+
```

Comment:
```suggestion
    void "spring-boot is not included if it's not added as a framework"(Language language, BuildTool buildTool) {
        when:
        String template = new BuildBuilder(beanContext, buildTool)
                .language(language)
                .features([SpringBootStarter.NAME])
                .render()
        BuildTestVerifier verifier = BuildTestUtil.verifier(buildTool, language, template)

        then:
        verifier.hasBuildPlugin("io.micronaut.application")
        verifier.hasBuildPlugin("com.github.johnrengelman.shadow")
        !verifier.hasBuildPlugin("org.springframework.boot")
        !verifier.hasBuildPlugin("io.spring.dependency-management")

        where:
        [language, buildTool] << [[Language.JAVA], BuildTool.valuesGradle()].combinations()
    }

```

Is it worth adding a test to check when framework is not set, that the features do not get used?

Reply:
I don't think it is necessary. All these java classes are in the test classpath. We test that io.micronaut.application is not applied thus it is testing that if you use a framework different than the one selected the default features of that framework are not applied.